File: src-all/Python/S0163-missing-ranges.py
"""
Given a sorted integer array nums, where the range of elements are in the inclusive range [lower, upper], return its missing ranges.

Example:

Input: nums = [0, 1, 3, 50, 75], lower = 0 and upper = 99,
Output: ["2", "4->49", "51->74", "76->99"]


"""
from typing import List
class Solution:
    def findMissingRanges(self, nums: List[int], lower: int, upper: int) -> List[str]:
        
        pass


if __name__ == '__main__':
    assert Solution().findMissingRanges(0) == 0

File: src-all/Rust/S0163-missing-ranges.rs
/*
Given a sorted integer array nums, where the range of elements are in the inclusive range [lower, upper], return its missing ranges.

Example:

Input: nums = [0, 1, 3, 50, 75], lower = 0 and upper = 99,
Output: [&quot;2&quot;, &quot;4->49&quot;, &quot;51->74&quot;, &quot;76->99&quot;]


*/
pub struct Solution {}
impl Solution {
    pub fn find_missing_ranges(nums: Vec<i32>, lower: i32, upper: i32) -> Vec<String> {
        
    }
}

fn main() {
  assert_eq!(0, Solution::find_missing_ranges(0));
  println!("Pass test cases!");
}
